### PR TITLE
feat: move to fork of `plausible-analytics`

### LINKED ISSUE #5: `enableAutoOutboundTracking` breaks `target="_blank"`
With autoOutboundTracking enabled Links marked with target="_blank" don't open in a new tab anymore.

### PR DESCRIPTION
**Pull request by Barbapapazes**
Hello,

I move this module to a fork of `plausible-tracker` to take back control of development. If I do correctly my job, there is no breaking change.

fix #5

### CODE REVIEW

**Comment by Barbapapazes** [comment]
For the next major version, we could simplify options.

**Comment by johannschopplich** [comment]
Lovely! Will take a deeper look into it later. Thanks in advance. 🙂

File: src/module.ts [hunk 1]
```diff
@@ -70,22 +90,29 @@ export default defineNuxtModule<ModuleOptions>({
...
     trackLocalhost: false,
     domain: '',
     apiHost: 'https://plausible.io',
     autoPageviews: true,
     autoOutboundTracking: false,
+    blackListedDomains: ['localhost'],
+    logIgnored: false,
   },
-  setup(options, nuxt) {
+  setup(userOptions, nuxt) {
```
Comment (Barbapapazes) [inline src/module.ts:103]:
```suggestion
  setup(options, nuxt) {
```

File: src/module.ts [hunk 2]
```diff
@@ -70,22 +90,29 @@ export default defineNuxtModule<ModuleOptions>({
...
     const { resolve } = createResolver(import.meta.url)
 
     // Add module options to public runtime config
     nuxt.options.runtimeConfig.public.plausible = defu(
       nuxt.options.runtimeConfig.public.plausible as Required<ModuleOptions>,
-      options,
+      userOptions,
```
Comment (Barbapapazes) [inline src/module.ts:109]:
```suggestion
      options,
```

File: src/module.ts [hunk 3]
```diff
@@ -70,22 +90,29 @@ export default defineNuxtModule<ModuleOptions>({
...
     )
 
+    if (!userOptions.enabled) {
```
Comment (Barbapapazes) [inline src/module.ts:112]:
```suggestion
    if (!options.enabled) {
```

File: src/module.ts [hunk 4]
```diff
@@ -101,5 +128,25 @@ export default defineNuxtModule<ModuleOptions>({
...
       mode: 'client',
     })
+
+    /**
+     * Split plugins to reduce bundle size.
+     *
+     * Use user options to install the plugin even if runtime config does not enable it. This allow users to enable the plugin later on.
+     */
+
+    if (userOptions.autoPageviews) {
```
Comment (Barbapapazes) [inline src/module.ts:138]:
```suggestion
    if (options.autoPageviews) {
```

File: src/module.ts [hunk 5]
```diff
@@ -101,5 +128,25 @@ export default defineNuxtModule<ModuleOptions>({
...
+      addPlugin({
+        src: resolve('runtime/plugin-auto-pageviews.client'),
+        mode: 'client',
+      })
+    }
+
+    if (userOptions.autoOutboundTracking) {
```
Comment (Barbapapazes) [inline src/module.ts:145]:
```suggestion
    if (options.autoOutboundTracking) {
```

**Comment by Barbapapazes** [comment]
Maybe we should update the compatibility to 3.10  since I use the dependsOn for plugins (https://github.com/nuxt/nuxt/pull/25409)

**Comment by johannschopplich** [comment]
As much as I like `dependsOn`, I don't think this breaking change is worth it. I know a handful of clients who haven't upgraded to newer Nuxt versions yet. I would much prefer providing users with your Plausible tracker fork than requiring them to upgrade to the latest Nuxt version as well. What do you think?

Can't we just use one plugin file and `await import` the functions, which are a separate plugin at the moment? Is the build chunk worth it?

Maybe migrate to `dependsOn` usage when it got more mainstream...

**Comment by Barbapapazes** [comment]
> As much as I like `dependsOn`, I don't think this breaking change is worth it. I know a handful of clients who haven't upgraded to newer Nuxt versions yet. I would much prefer providing users with your Plausible tracker fork than requiring them to upgrade to the latest Nuxt version as well. What do you think?
> 
> Can't we just use one plugin file and `await import` the functions, which are a separate plugin at the moment? Is the build chunk worth it?
> 
> Maybe migrate to `dependsOn` usage when it got more mainstream...
> 
> Having said that, maybe enforcing Nuxt 3.10 is OK. I haven't made up my mind yet. 😄 Your plugin split is much appreciated tho.

Will investigate but maybe, the `dependsOn` is useful when you depend on plugins from another module. Since it's an array, the order may be sufficient to initialize extensions after the main one.

**Comment by johannschopplich** [comment]
Great job, thanks a lot!

Note:
As far as I understand it, we can upgrade Nuxt, but if we would have to tighten compatibility, we would have to define it in the module meta:

```ts
compatibility: {
  nuxt: '^3',
},
```

**Pull request merged**